helpers/auth/authentication.py

```diff
@@ -134,14 +134,13 @@ def wrapper(*args, **kwargs):
                     except Exception:
                         raise GraphQLError("The database cannot be reached")
                     for value in response['values']:
-                        if user.email == value["email"]:
-                            if value['location'] and not user.location:
-                                user_location = value['location'][
-                                    'name'
-                                ] or "Nairobi"
-                                user.location = user_location
-                                user.save()
-                                check_and_add_location(user_location)
+                        if not user.location:
```

Comment:
@mnswaleh  Based on this implementation, the user location will still be `Null` if the `values` array from Andela api is empty as lines 137 to 143 will not be executed.